DOL.WHD.Section14c.Business/Services/AttachmentService.cs

```diff
@@ -28,19 +27,21 @@ public AttachmentService(IFileRepository fileRepository, IAttachmentRepository a
 
         public Attachment UploadAttachment(string applicationId, byte[] bytes, string fileName, string fileType)
         {
+            string FileEncryptKey = RandomString(40);
```

Comment:
This worries me for two reasons:  the key is limited to alphanumeric characters, and those characters are picked by a cryptographically insecure random number generator.

Can the `EncryptionKey` field in the database be a byte array?  If so, I would strongly encourage switching to that and using the [RNGCryptoServiceProvider.GetBytes()](https://msdn.microsoft.com/en-us/library/wb9c8c67(v=vs.110).aspx) method to generate the key bytes.  That would dramatically increase the [key space](https://en.wikipedia.org/wiki/Key_space_(cryptography)) and use a [cryptographically-secure random number generator](https://en.wikipedia.org/wiki/Cryptographically_secure_pseudorandom_number_generator)